Add options to control pulse duration on Climate_IR_LG Component

mhentschke opened the pull request:
## Description:
Added Parameters:
header_high
header_low
bit_high
bit_one_low
bit_zero_low

These parameters control the timings of the protocol used for LG AC Units. In Brazil, for example, some units use different than standard timings but the protocol is essentially the same.

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#963

## Checklist:
  - [X] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

mhentschke committed:
Added options to control pulse duration on Climate_IR_LG Component. This is usefull as some equipment from LG (Tested in Brazil AC unit) use different pulse durations in their protocol.

mhentschke committed:
Fixed C++ linting issues

mhentschke committed:
Fixed Python linting issues

mhentschke committed:
fixed spaces on parameters linting issue

mhentschke committed:
fixed spacing clint

glmnet commented:
are the const in the top of cpp still in use?

mhentschke committed:
Removed unused constants

mhentschke commented:
Indeed they aren't. I forgot to remove them. Thanks for pointing that out! :D

mhentschke committed:
Removed wrong spacing

mhentschke commented:
Not sure what tests need to be added to this. Can anybody help out on that matter?

glmnet commented:
the test is the component being included in any of the test_.yaml files so the code generation includes this cpp code to be compiled so it is checked, linted, etc.

This pr is good, docs are not. We merge them together

mhentschke committed:
Changed int to time period in all new fields

Changed cv._int to cv.positive_time_period_microseconds in the time definitions for the new options

glmnet commented:
ahh good. Didn't see that one, linter is now complaining the type, should suffix with `us` for microseconds.

mhentschke committed:
Fixed the time defaults

Time defaults fixed for Climate_IR_LG.

mhentschke commented:
Everything should be good now. Please let me know in case there's anything else needed to get this going :D
